src/server/game/Entities/Player/CollectionMgr.cpp

```diff
@@ -603,6 +603,58 @@ void CollectionMgr::AddItemAppearance(uint32 itemId, uint32 appearanceModId /*=
     AddItemAppearance(itemModifiedAppearance);
 }
 
+void CollectionMgr::AddTransmogSet(uint32 transmogSetId)
+{
+    std::vector<TransmogSetItemEntry const*> const* items = sDB2Manager.GetTransmogSetItems(transmogSetId);
+    if (!items)
+        return;
+
+    for (TransmogSetItemEntry const* item : *items)
+    {
+        ItemModifiedAppearanceEntry const* itemModifiedAppearance = sItemModifiedAppearanceStore.LookupEntry(item->ItemModifiedAppearanceID);
+        if (!itemModifiedAppearance)
+            continue;
+
+        AddItemAppearance(itemModifiedAppearance);
+    }
+}
+
+bool CollectionMgr::IsSetCompleted(uint32 transmogSetId) const
```

Comment:
https://gist.github.com/Shauren/6929981f075b8a40b6c822a209fd518a
Improved version with no temporary containers (this is what the client does too btw with ItemTransmogrificationSlots)